snapm/manager/_boot.py

```diff
@@ -88,6 +210,76 @@ def _get_machine_id():
     return machine_id
 
 
+def get_device_path(identifier: str, by_type: str) -> Optional[str]:
+    """
+    Translates a filesystem UUID or label to its corresponding device path
+    using the blkid command.
+
+    Args:
+        identifier: The UUID or label of the filesystem.
+        by_type: Specifies the type of identifier. Must be "uuid" or "label".
+
+    Returns:
+        The device path (e.g., '/dev/sda1') if found, otherwise None.
+
+    Raises:
+        ValueError: If 'identifier' is empty or 'by_type' is not 'uuid' or 'label'.
+        FileNotFoundError: If the 'blkid' command is not found on the system.
+        subprocess.CalledProcessError: If the 'blkid' command exits with a non-zero status
+                                       due to reasons other than the identifier not being found
+                                       (e.g., permission issues).
+        Exception: For any other unexpected errors during command execution or parsing.
+    """
+    if not identifier:
+        raise ValueError("Identifier cannot be an empty string.")
+    if by_type not in ["uuid", "label"]:
+        raise ValueError("Invalid 'by_type'. Must be 'uuid' or 'label'.")
+
+    try:
+        command = ["sudo", "blkid"]
+
+        if by_type == "uuid":
+            command.extend(["--uuid", identifier])
+        else:  # by_type == "label"
+            command.extend(["-L", identifier])
+
+        # Execute the command
+        result = subprocess.run(command, capture_output=True, text=True, check=True)
+
+        # The output of 'blkid --uuid <UUID>' or 'blkid -L <LABEL>' is simply the device path
+        # if found. If not found, it returns a non-zero exit code, which
+        # 'check=True' converts into a CalledProcessError.
+        device_path = result.stdout.strip()
+
+        if device_path:
+            return device_path
+        else:
+            # This case should ideally not be reached if check=True is working as expected
+            # for "not found" scenarios, but it's a safeguard.
+            return None
```

Comment:
You can drop the `else` here and de-indent the `return None` - since the other branch of the `if` ends in a return statement.